### PR TITLE
Launch gz_spawn_model from xml

### PR DESCRIPTION
**Pull request by caguero**
# 🎉 New feature

Part of https://github.com/gazebosim/ros_gz/issues/552

## Summary

This patch lets you spawn a gz sim model from an XML file.

### How to test it?

Launch gz sim:
```
ros2 launch ros_gz_sim gz_sim.launch.py gz_args:=empty.sdf
```

Hit play.

Use the provided `gz_spawn_model.launch` to spawn a vehicle (change `file` accordingly):
```
ros2 launch ros_gz_sim gz_spawn_model.launch world:=empty file:=/home/caguero/ros_gz_ws/src/ros_gz/ros_gz_sim_demos/models/vehicle/model.sdf name:=my_vehicle x:=5.0 y:=5.0 z:=0.5
```

A vehicle should be loaded into the scene.

Confirm that no container has been created:
```
caguero@cold:~/ros_gz_ws$ ros2 component list

```
## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: launch_gz/launch_gz/actions/gz_spawn_model.py [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+        xml_string: SomeSubstitutionsType,
+        topic: SomeSubstitutionsType,
+        name: SomeSubstitutionsType,
+        allow_renaming: SomeSubstitutionsType,
+        x: SomeSubstitutionsType,
+        y: SomeSubstitutionsType,
+        z: SomeSubstitutionsType,
+        roll: SomeSubstitutionsType,
+        pitch: SomeSubstitutionsType,
+        yaw: SomeSubstitutionsType,
```
Comment (azeey) [inline launch_gz/launch_gz/actions/gz_spawn_model.py]: Can these be made optional similar to my other [feedback](https://github.com/gazebosim/ros_gz/pull/528/files#r1610747230)?

File: ros_gz_sim/launch/gz_spawn_model.launch [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    xml_string="$(var xml_string)"
+    topic="$(var topic)"
+    name="$(var name)"
+    allow_renaming="$(var allow_renaming)"
+    x="$(var x)"
+    y="$(var y)"
+    z="$(var z)"
+    roll="$(var roll)"
+    pitch="$(var pitch)"
+    yaw="$(var yaw)"
```
Comment (ahcorde) [inline ros_gz_sim/launch/gz_spawn_model.launch]: This looks wrong, isn't it?
```suggestion
    yaw="$(var yaw)">
```

**Review by ahcorde — CHANGES_REQUESTED** [review]
resolve conflicts too

File: launch_gz/launch_gz/actions/gz_spawn_model.py (hunk 1 above)
Comment (caguero) [reply launch_gz/launch_gz/actions/gz_spawn_model.py]: Changed in [04c9dc8](https://github.com/gazebosim/ros_gz/pull/551/commits/04c9dc8585abac796b68987a5f25c78677c656a4).

File: ros_gz_sim/launch/gz_spawn_model.launch (hunk 2 above)
Comment (caguero) [reply ros_gz_sim/launch/gz_spawn_model.launch]: Typo fixed in [04c9dc8](https://github.com/gazebosim/ros_gz/pull/551/commits/04c9dc8585abac796b68987a5f25c78677c656a4).

**Review by azeey — APPROVED** [review]
LGTM!

**Review by ahcorde — APPROVED** [review]

**Pull request merged**